libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/RouteUtils.java

```diff
@@ -75,12 +76,25 @@ public static boolean isNewRoute(@Nullable RouteProgress previousRouteProgress,
    * checks if the arrival meter threshold has been hit.
    *
    * @param routeProgress the current route progress
+   * @param milestone     the current milestone from the MilestoneEventListener
    * @return true if in arrival state, false if not
    * @since 0.8.0
    */
-  public static boolean isArrivalEvent(@NonNull RouteProgress routeProgress) {
-    return (upcomingStepIsArrival(routeProgress) || currentStepIsArrival(routeProgress))
-      && routeProgress.currentLegProgress().distanceRemaining() <= METERS_REMAINING_TILL_ARRIVAL;
+  public static boolean isArrivalEvent(@NonNull RouteProgress routeProgress, @NonNull Milestone milestone) {
+    if (!(milestone instanceof BannerInstructionMilestone)) {
+      return false;
+    }
+    if (upcomingStepIsArrival(routeProgress) || currentStepIsArrival(routeProgress)) {
```

Comment:
Can you explain what this is doing/why? Maybe this can be refactored to make it more clear